### PR TITLE
fix weight drop and add tests

### LINKED ISSUE #264: apply_weight_drop required changes to integrate mxnet 1.3
our existing implementation on apply_weight_drop relied on a non-public attribute in gluon RNN blocks that were removed apache/incubator-mxnet#11482 when making the RNN blocks hybrid, and it does not have test, which will prevent us from integrating with mxnet 1.3. We need:
- [ ] fix apply_weight_drop and upgrade the mxnet dependency to allow 1.3 integration.
- [ ] add test for apply_weight_drop
- [ ] bump up minor version to 0.4 which relies on 1.3.

### PR DESCRIPTION
**Pull request by szha**
## Description ##
fix weight drop with new mxnet version

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] update mxnet dependency to the latest nightly
- [x] fix weight drop
- [x] add weight drop tests

## Comments ##
- resolves #264

### CODE REVIEW

File: tests/unittest/test_models.py [hunk 1]
```diff
@@ -207,3 +206,68 @@ def test_big_rnn_model_share_params():
     logits, hidden = eval_model(x, hidden)
     assert logits.shape == (seq_len, batch_size, vocab_size)
     mx.nd.waitall()
+
+def test_weight_drop():
```
Comment (szhengac) [inline tests/unittest/test_models.py]: Perhaps we also need to test whether the gradient is correct. We can do it by asserting the new parameter.

Comment (szha) [reply tests/unittest/test_models.py]: added

**Comment by mli** [comment]
Job PR-268/5 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-268/5/index.html

File: gluonnlp/model/utils.py [hunk 2]
```diff
@@ -19,21 +19,25 @@
...
     """Apply weight drop to the parameter of a block.
 
     Parameters
     ----------
     block : Block or HybridBlock
         The block whose parameter is to be applied weight-drop.
-    local_param_name : str
-        The parameter name used on the block. such as 'weight'.
+    local_param_regex : str
+        The regex for parameter names used in the self.params.get(), such as 'weight'.
```
Comment (eric-haibin-lin) [inline gluonnlp/model/utils.py]: Can we update the doc with an example of using regex instead of exact match? such as ".*h2h_weight"

**Comment by mli** [comment]
Job PR-268/10 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-268/10/index.html
